HTMLtoMarkdown_Prewrite.cchtml

```diff
@@ -0,0 +1,127 @@
+<h1>HTML to Markdown Conversion</h1>
+<h3>Sprint 2 Developers:</h3>
+<p>Kenzie Ellenberger: <a
+        href="mailto:mde168@msstate.edu">mde168@msstate.edu</a></p>
+<p>Clark Hensley: <a href="mailto:ch3136@msstate.edu">ch3136@msstate.edu</a>
+</p>
+<h3><strong>1. Feature to be Added</strong></h3>
+<div>The feature we will be adding is the conversion from HTML to Markdown
+    within the CodeChat Editor. This will be implemented to convert the HTML
+    produced on save for files to be stored in Markdown within the source code.
+    Converting the HTML to Markdown will make the source code easier for the
+    user to read from within an IDE or anytime when viewing the source
+    code.&nbsp;</div>
+<h3><strong>2. Repositories the New Code will be Added to</strong></h3>
+<p>The new code will first go through the forked&nbsp;<a
+        href="https://github.com/ClarkHensley/CodeChat_Editor">CodeChat_Editor
+    </a>repository with the owner ClarkHensley on Github. A pull request will
+    be made to the original repository, also <a
+        href="https://github.com/bjones1/CodeChat_Editor">CodeChat_Editor
+    </a>with the owner bjones1. The code will then be merged into this main
+    repository.&nbsp;</p>
+<h3>3. Libraries, languages, and interfaces our code must work with.</h3>
+<p>The programming language we will be using to implement this new feature is
+    Typescript. The markup languages we will have to work with while
+    implementing this feature are HTML and Markdown. To convert the HTML to
+    Markdown, the <a href="https://github.com/mixmark-io/turndown">Turndown
+    </a>library will be used. This code will be added within the save function
+    for the CodeChat Editor so it will need to work with the methods
+    surrounding that functionality.&nbsp;</p>
+<h3>4. Series of steps to complete this feature</h3>
+<p>Step 1: Add the Turndown library to the project. Do this by installing it
+    with npm with the following command:</p>
+<p><code> npm install turndown </code></p>
+<p>This will add the library to the list of dependencies within the
+    package.json file. Now on npm install or npm update, this package/library
+    will be installed.&nbsp;</p>
+<p>Step 2: Load the library into the <a
+        href="client/src/CodeChatEditor.mts">CodeChatEditor.mts</a> with the <a
+        href="https://www.typescriptlang.org/docs/handbook/modules.html">require()</a>
+    function. This imports the module into the file, and allows an instance of
+    the class to be created. Once the class is instantiated, this variable can
+    be used to use the Turndown library functions. The code to instantiate the
+    module is shown below.&nbsp;</p>
+<p><code> const TurndownService = require("turndown");
```

Comment:
I'm using ESM, not NPM's old require system, so this should be an`import` statement instead.